libs/dal/src/repositories/notification-template/notification-template.entity.ts

```diff
@@ -25,6 +26,19 @@ import type { EnvironmentId } from '../environment';
 import type { ChangePropsValueType } from '../../types';
 import { JSONSchema } from 'json-schema-to-ts';
 
+type WorkflowOptionsPreference = {
+  workflow: {
+    defaultValue: boolean;
+    readOnly: boolean;
+  };
+  channels: {
+    [key in (typeof ChannelTypeEnum)[keyof typeof ChannelTypeEnum]]?: {
```

Comment:
Readability improvement:

```suggestion
    [`$ChannelTypeEnum`]?: {
```

Reply:
I tried this but got:
```
A computed property name in a type literal must refer to an expression whose type is a literal type or a 'unique symbol' type.ts(1170)
'ChannelTypeEnum' will evaluate to '[object Object]' when stringified.
```

Reply:
Interesting, I would've expected Typescript to handle that! Let's go with the original then.